core/vm/common.go

```diff
@@ -60,7 +60,7 @@ func getData(data []byte, start uint64, size uint64) []byte {
 	if end > length {
 		end = length
 	}
-	return common.RightPadBytes(data[start:end], int(size))
+	return common.RightPadBytes(data[start:end], size)
```

Comment:
I think you changed `RightPadBytes` because of this use, but there must be a better way. Maybe the `size` parameter should have type `int` here?

I can see that `getData` is used with a non-constant `size` argument in some precompiles.

Reply:
Semantically, I think it is more correct to RightPadBytes with an unsigned size. 

I'm a bit annoyed that `len` is `int` and not `uint` in the first place...

Reply:
```suggestion
	return common.RightPadBytes(data[start:end], size)
```